src/app/services/courses-store.service.ts

```diff
@@ -1,42 +1,165 @@
-import { Injectable } from '@angular/core';
+import { Injectable } from "@angular/core";
+import { Course, newCourse } from "@app/shared/models/course.model";
+import { BehaviorSubject, catchError, Observable, tap } from "rxjs";
+import { CoursesService } from "./courses.service";
+import { Author } from "@app/shared/models/author.model";
 
 @Injectable({
-    providedIn: 'root'
+  providedIn: "root",
 })
 export class CoursesStoreService {
-    getAll(){
-        // Add your code here
-    }
+  private courses$$: BehaviorSubject<Course[]> = new BehaviorSubject<Course[]>(
+    []
+  );
+  private loading$$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(
+    false
+  );
+  private authors$$: BehaviorSubject<Author[]> = new BehaviorSubject<Author[]>(
+    []
+  );
+  public courses$: Observable<Course[]> = this.courses$$.asObservable();
+  public loading$: Observable<boolean> = this.loading$$.asObservable();
+  public authors$: Observable<Author[]> = this.authors$$.asObservable();
 
-    createCourse(course: any) { // replace 'any' with the required interface
-        // Add your code here
-    }
+  constructor(private coursesService: CoursesService) {}
+  getAll() {
+    this.loading$$.next(true);
+    this.coursesService
+      .getAll()
+      .pipe(
+        tap((response) => {
+          this.courses$$.next(response.result);
+          this.loading$$.next(false);
+        }),
+        catchError((error) => {
+          this.loading$$.next(false);
+          console.error("Error getting courses from server:", error);
 
-    getCourse(id: string) {
-        // Add your code here
-    }
+          return [];
+        })
+      )
+      .subscribe();
```

Comment:
What do you subscribe to here exactly?
Probably you want to use the returned data (e.g. assign it somewhere) from the observable if you make the subscription here. Please make an update to the other occurrences, too.